pkg/filenotify/poller.go

```diff
@@ -123,24 +123,56 @@ func (w *filePoller) Close() error {
 	return nil
 }
 
-// sendEvent publishes the specified event to the events channel
-func (w *filePoller) sendEvent(e fsnotify.Event, chClose <-chan struct{}) error {
+var errRetry = errors.New("retry send again")
+
+func (w *filePoller) trySendEvent(e fsnotify.Event, chClose <-chan struct{}) error {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if w.closed {
+		return errPollerClosed
+	}
 	select {
 	case w.events <- e:
+		return nil
 	case <-chClose:
 		return fmt.Errorf("closed")
+	default:
+		return errRetry
 	}
-	return nil
 }
 
-// sendErr publishes the specified error to the errors channel
-func (w *filePoller) sendErr(e error, chClose <-chan struct{}) error {
+// sendEvent publishes the specified event to the events channel
+func (w *filePoller) sendEvent(e fsnotify.Event, chClose <-chan struct{}) error {
+	for {
```

Comment:
Maybe if `w.close` was a channel we could avoid this loop. Not quite sure how often you would need to retry but if it blocks often this is bit inefficient.